kbase-extension/static/kbase/js/common/ui.js

```diff
@@ -241,38 +263,73 @@ define([
         modalDialogNode = modalNode.querySelector('.modal');
         $(modalDialogNode).modal('show');
         return new Promise(function (resolve, reject) {
-            modalDialogNode.querySelector('[data-element="cancel"]').addEventListener('click', function (e) {
-                confirmNode.parentElement.removeChild(confirmNode);
-                resolve({
-                    action: 'cancel'
+            modalDialogNode
+                .querySelector('[data-element="cancel"]')
+                .addEventListener('click', function (e) {
+                    confirmNode.parentElement.removeChild(confirmNode);
+                    resolve({
+                        action: 'cancel',
+                    });
                 });
-            });
             args.buttons.forEach(function (btn) {
-                modalDialogNode.querySelector('[data-element="' + btn.action + '"]').addEventListener('click', function (e) {
-                    try {
-                        var result = btn.handler(e);
-                        if (result) {
-                            $(modalDialogNode).modal('hide');
-                            confirmNode.parentElement.removeChild(confirmNode);
-                            resolve({
-                                action: btn.action,
-                                result: result
-                            });
+                modalDialogNode
+                    .querySelector('[data-element="' + btn.action + '"]')
+                    .addEventListener('click', function (e) {
+                        try {
+                            var result = btn.handler(e);
+                            if (result) {
+                                $(modalDialogNode).modal('hide');
+                                confirmNode.parentElement.removeChild(confirmNode);
+                                resolve({
+                                    action: btn.action,
+                                    result: result,
+                                });
+                            }
+                        } catch (ex) {
+                            reject(ex);
                         }
-                    } catch (ex) {
-                        reject(ex);
-                    }
-                });
+                    });
             });
 
             modalDialogNode.addEventListener('hide.bs.modal', function (e) {
                 resolve({
-                    action: 'cancel'
+                    action: 'cancel',
                 });
             });
         });
     }
 
+    /**
```

Comment:
Slightly modified loading function here. Added some comments and converted from `var` to `let`.